src/connector/src/sink/encoder/proto.rs

```diff
@@ -364,18 +409,61 @@ fn encode_field<D: MaybeData>(
                     Ok(Value::Message(message.transcode_to_dynamic()))
                 })?
             }
+            (false, Kind::String) if is_big_query => {
```

Comment:
It seems that the newly added `custom_proto_type` is only used to generate the `is_big_query` flag, and the flag is only used to control the logic when seeing different datatypes. If so, instead of adding this new parameter, we can just add new methods like `on_timestamptz`, `on_jsonb` ... to the `MaybeData` trait and then call the corresponding trait methods here. And then for bigquery we can implement its own `MaybeData` with new customized logic and then pass it to the encoder.

Reply:
There's rw and proto type match , bigquery has some special matches (many of which are converted to string) that require `is_big_query `to determine whether the match holds or not